### PR TITLE
Support static reports in mobile UCR

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Allows `StaticReportConfiguration`s to be used for mobile UCR.

Best reviewed commit-by-commit
@nickpell 
cc @esoergel

### CODE REVIEW

File: corehq/apps/userreports/tests/test_get_report_configs.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            return []
+
+        with patch('corehq.apps.userreports.models.get_docs', new=get_docs_mock):
+            with override_settings(STATIC_UCR_REPORTS=[self.get_path('static_report_config', 'json')]):
+                configs = get_report_configs(
+                    [dynamic_report_id, StaticReportConfiguration.get_doc_id('example', 'a-custom-report', False)],
+                    'example'
+                )
+                self.assertEqual(len(configs), 2)
+
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_get_report_configs.py]: W391 blank line at end of file

File: corehq/apps/app_manager/suite_xml/features/mobile_ucr.py [hunk 2]
```diff
@@ -99,19 +100,19 @@ def _get_select_details(config):
...
+        for chart_config in config.report(domain).charts:
             if isinstance(chart_config, MultibarChartSpec):
                 graph_config = config.graph_configs.get(chart_config.chart_id, ReportGraphConfig())
 
                 def _column_to_series(column):
                     return Series(
                         nodeset=(
                             "instance('reports')/reports/report[@id='{}']/rows/row[@is_total_row='False']{}"
-                            .format(config.uuid, _MobileSelectFilterHelpers.get_data_filter_xpath(config))),
+                            .format(config.uuid, _MobileSelectFilterHelpers.get_data_filter_xpath(config, domain))),
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/features/mobile_ucr.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/app_manager/views/modules.py [hunk 3]
```diff
@@ -184,7 +185,8 @@ def _report_to_config(report):
             'filter_structure': report.filters,
         }
 
-    all_reports = ReportConfiguration.by_domain(app.domain)
+    all_reports = ReportConfiguration.by_domain(app.domain) + \
+                  StaticReportConfiguration.by_domain(app.domain)
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: E127 continuation line over-indented for visual indent

**Comment by nickpell** [comment]
:+1: when :white_check_mark:

**Comment by nickpell** [comment]
```
======================================================================
ERROR: setUpClass (corehq.apps.app_manager.tests.test_report_config.ReportFiltersSuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/app_manager/tests/test_report_config.py", line 136, in setUpClass
    fixture, = report_fixture_generator(cls.user, '2.0')
  File "/mnt/corehq/apps/app_manager/fixtures/mobile_ucr.py", line 42, in __call__
    reports_elem.append(self._report_config_to_fixture(report_config, user))
  File "/mnt/corehq/apps/app_manager/fixtures/mobile_ucr.py", line 60, in _report_config_to_fixture
    report = get_report_config(report_config.report_id, user.domain)[0]
  File "/mnt/corehq/apps/userreports/models.py", line 645, in get_report_config
    config = get_report_configs([config_id], domain)[0]
  File "/mnt/corehq/apps/userreports/models.py", line 629, in get_report_configs
    get_docs(ReportConfiguration.get_db(), dynamic_report_config_ids)
  File "submodules/dimagi-utils-src/dimagi/utils/couch/bulk.py", line 85, in get_docs
    url = db.uri + '/_all_docs'
  File "/vendor/lib/python2.7/site-packages/mock.py", line 700, in __getattr__
    raise AttributeError("Mock object has no attribute %r" % name)
AttributeError: Mock object has no attribute 'uri'
```

File: corehq/apps/app_manager/tests/test_report_fixtures_provider.py [hunk 4]
```diff
@@ -30,10 +31,9 @@ def test_report_fixtures_provider(self):
         )
         user = Mock()
 
-        with patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportConfiguration') as report_config_patch, \
-                patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
+        with mock_report_configuration_get({report_id: MAKE_REPORT_CONFIG('test_domain', report_id)}), \
+            patch('corehq.apps.app_manager.fixtures.mobile_ucr.ReportFactory') as report_factory_patch:
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_report_fixtures_provider.py]: E125 continuation line with same indent as next logical line

**Comment by NoahCarnahan** [comment]
Thanks. Pushed a commit that should fix the tests.

**Comment by NoahCarnahan** [comment]
✅

**Pull request merged**